server/src/main/scala/org/scassandra/server/cqlmessages/types/ColumnType.scala

```diff
@@ -15,21 +15,26 @@
  */
 package org.scassandra.server.cqlmessages.types
 
-import akka.util.ByteIterator
+import akka.util.{ByteString, ByteIterator}
 import com.typesafe.scalalogging.LazyLogging
-import org.apache.cassandra.serializers.TypeSerializer
 import org.scassandra.cql._
 import org.scassandra.server.cqlmessages.ProtocolVersion
 
+object WriteScope extends Enumeration {
+  val Standalone, InCollection = Value
+}
+
 abstract class ColumnType[T](val code : Short, val stringRep: String) extends LazyLogging {
-  def readValue(byteIterator : ByteIterator, protocolVersion: ProtocolVersion) : Option[T]
-  def writeValue(value : Any) : Array[Byte]
-  def writeValueInCollection(value: Any) : Array[Byte] = ???
-  def readValueInCollection(byteIterator: ByteIterator) : T = ???
-  def serializer: TypeSerializer[T] = ???
-  def convertToCorrectCollectionTypeForList(list: Iterable[_]) : List[T] = ???
-  def convertToCorrectCollectionTypeForSet(set: Iterable[_]) : Set[T] = convertToCorrectCollectionTypeForList(set).toSet
-  def convertToCorrectJavaTypeForSerializer(value: Any): T
+  implicit val byteOrder = java.nio.ByteOrder.BIG_ENDIAN
+  def readValue(byteIterator : ByteIterator)(implicit protocolVersion: ProtocolVersion) : Option[T]
+  def writeValue(value : Any)(implicit protocolVersion: ProtocolVersion) : Array[Byte]
+  def readValueWithLength(byteIterator: ByteIterator)(implicit protocolVersion: ProtocolVersion): Option[T] = {
```

Comment:
Since depending on protocolVersion and context (whether or not value will be part of a collection) matters when encoding and decoding length I've decided to divorce length from read and writeValue.  Instead the byteIterator passed into readValue will only encompass the bytes for that data (so code can make an assumption about length).  Additionally, code that calls writeValue must encode the length on the outside now (CqlProtocolHelper.withLength provided for this, and collection serializers do this too).  readValueWithLength is a convenience method which takes a ByteIterator and gets the necessary slice of data based on read int length and passes the the slice to readValue (I'll document this).